Lesson16/Dilek/script.js

```diff
@@ -0,0 +1,498 @@
+/*
+1. Sum Array Elements with a For Loop
+   - Define a function `sumArray(numbers)` that uses a for loop
+     to sum all elements in an array of numbers.
+   - Log the final sum.
+*/
+
+const arrEx1 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+function sumArray(numbers) {
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    sum = sum + numbers[i];
+  }
+  console.log(sum);
+}
+
+sumArray(arrEx1);
+console.log('-------------------');
+
+/*
+2. Find Maximum Number in an Array
+   - Define a function `findMax(numbers)` that uses a for loop to iterate
+     through an array and find the largest value.
+   - Log the largest value.
+*/
+
+const arrEx2 = [1, 2, 30, 4, 5, 600, 7, 8, 9, 10];
+
+function findMax(numbers) {
+  let max = numbers[0];
+
+  // for (const number of numbers) {
+  //   if (number > max) {
+  //     max = number;
+  //   }
+  //     console.log(number, max);
+  // }
+
+  for (let i = 1; i < numbers.length; i++) {
+    if (numbers[i] > max) {
+      max = numbers[i];
+    }
+  }
+  return max;
+}
+
+console.log(findMax(arrEx2));
+console.log('-------------------');
+
+/*
+3. Count Odd and Even Numbers
+   - Define a function `countOddEven(numbers)` that loops through an array
+     of numbers and counts how many are odd and how many are even.
+   - Log the counts in the format: 'Odd: X, Even: Y'
+*/
+
+const arrEx3 = [1, 2, 30, 4, 5, 600, 7, 8, 9, 10];
+function countOddEven(numbers) {
+  let oddTotal = 0;
+  let evenTotal = 0;
+  for (number of numbers) {
```

Comment:
Don't omit keyword 'const' in the for...of / for...in loops. You need to specify it to correctly declare a variable.
```suggestion
  for (const number of numbers) {
```